doc/rst/developer/bestPractices/README.rst

```diff
@@ -1,57 +0,0 @@
-=====================
```

Comment:
If you do still want to delete this file, it might be worth doing a repo search for other READMEs or other places that link to it (noticed by glancing over the Potpourri doc)

Reply:
IMO index.rst is about as obvious as README.rst for somebody browsing a file list. If we think README.rst is important, I'd rather change it to just README and have 'see index.rst' in it.

The only references to bestPractices/README.rst that I find are in spell checking related files and I think it's OK not to update these in this PR.

Reply:
I know that I can `include` the README.rst from index.rst but I don't like that strategy for two reasons:
 1. It is awkward to fix the ensuing "Found Document Not in Toctree!" errors that Sphinx will give
 2. The markup here is structured in a way that makes sense for index.rst but would not make sense for README.rst

Reply:
> IMO index.rst is about as obvious as README.rst for somebody browsing a file list. If we think README.rst is important, I'd rather change it to just README and have 'see index.rst' in it.

Does Github pick up index.rst to automatically display when you surf to a directory via its UI?  If not, that's what we're losing by moving away from README.rst

Reply:
It doesn't. One thing I have made here is https://github.com/mppf/chapel/tree/compiler-internals-adj/doc/rst/developer/bestPractices which has a README.md than links to index.rst but when we view index.rst in github, the links are all broken, but maybe that's good enough for this?

Reply:
OK, I've done that, but we can't check the link yet until this PR is merged.